File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/entity/WorkflowExecutionMeta.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package com.aiflow.entity;

import com.aiflow.common.State;
import com.aiflow.proto.Message.WorkflowExecutionProto;
import com.aiflow.proto.MetadataServiceOuterClass.WorkFlowExecutionListProto;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class WorkflowExecutionMeta {

    private Long uuid;
    private String name;
    private State executionState;
    private Long projectId;
    private Map<String, String> properties;
    private Long startTime;
    private Long endTime;
    private String logUri;
    private String workflowJson;
    private String signature;

    public WorkflowExecutionMeta() {
    }

    public WorkflowExecutionMeta(Long uuid, String name, State executionState, Long projectId, Map<String, String> properties, Long startTime, Long endTime, String logUri, String workflowJson, String signature) {
        this.uuid = uuid;
        this.name = name;
        this.executionState = executionState;
        this.projectId = projectId;
        this.properties = properties;
        this.startTime = startTime;
        this.endTime = endTime;
        this.logUri = logUri;
        this.workflowJson = workflowJson;
        this.signature = signature;
    }

    public Long getUuid() {
        return uuid;
    }

    public void setUuid(Long uuid) {
        this.uuid = uuid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public State getExecutionState() {
        return executionState;
    }

    public void setExecutionState(State executionState) {
        this.executionState = executionState;
    }

    public Long getProjectId() {
        return projectId;
    }

    public void setProjectId(Long projectId) {
        this.projectId = projectId;
    }

    public Map<String, String> getProperties() {
        return properties;
    }

    public void setProperties(Map<String, String> properties) {
        this.properties = properties;
    }

    public Long getStartTime() {
        return startTime;
    }

    public void setStartTime(Long startTime) {
        this.startTime = startTime;
    }

    public Long getEndTime() {
        return endTime;
    }

    public void setEndTime(Long endTime) {
        this.endTime = endTime;
    }

    public String getLogUri() {
        return logUri;
    }

    public void setLogUri(String logUri) {
        this.logUri = logUri;
    }

    public String getWorkflowJson() {
        return workflowJson;
    }

    public void setWorkflowJson(String workflowJson) {
        this.workflowJson = workflowJson;
    }

    public String getSignature() {
        return signature;
    }

    public void setSignature(String signature) {
        this.signature = signature;
    }

    @Override
    public String toString() {
        return "WorkflowExecutionMeta{" +
                "uuid=" + uuid +
                ", name='" + name + '\'' +
                ", executionState=" + executionState +
                ", projectId=" + projectId +
                ", properties=" + properties +
                ", startTime=" + startTime +
                ", endTime=" + endTime +
                ", logUri='" + logUri + '\'' +
                ", workflowJson='" + workflowJson + '\'' +
                ", signature='" + signature + '\'' +
                '}';
    }

    public static WorkflowExecutionMeta buildWorkflowExecutionMeta(WorkflowExecutionProto workflowExecutionProto) {
        return workflowExecutionProto == null ? null : new WorkflowExecutionMeta(workflowExecutionProto.getUuid(),
                workflowExecutionProto.getName(),
                State.valueOf(workflowExecutionProto.getExecutionState().name()),
                workflowExecutionProto.getProjectId().getValue(),
                workflowExecutionProto.getPropertiesMap(),
                workflowExecutionProto.getStartTime().getValue(),
                workflowExecutionProto.getEndTime().getValue(),
                workflowExecutionProto.getLogUri().getValue(),
                workflowExecutionProto.getWorkflowJson().getValue(),
                workflowExecutionProto.getSignature().getValue());
    }

    public static List<WorkflowExecutionMeta> buildWorkflowExecutionMetas(WorkFlowExecutionListProto workFlowExecutionListProto) {
        if (workFlowExecutionListProto == null) {
            return null;
        } else {
            List<WorkflowExecutionMeta> workflowExecutionMetas = new ArrayList<>();
            for (WorkflowExecutionProto workflowExecutionProto : workFlowExecutionListProto.getWorkflowExecutionsList()) {
                workflowExecutionMetas.add(buildWorkflowExecutionMeta(workflowExecutionProto));
            }
            return workflowExecutionMetas;
        }
    }
}

File: flink-ai-flow/ai_flow/meta/workflow_execution_meta.py
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
#

from ai_flow.util.json_utils import Jsonable
from ai_flow.common.properties import Properties
from typing import Text
from ai_flow.meta.job_meta import State


class WorkflowExecutionMeta(Jsonable):
    """define workflow execution meta"""

    def __init__(self,
                 name: Text,
                 execution_state: State,
                 project_id: int = None,
                 properties: Properties = None,
                 start_time: int = None,
                 end_time: int = None,
                 log_uri: Text = None,
                 workflow_json=None,
                 signature=None,
                 uuid: int = None
                 ) -> None:
        """

        :param name: the workflow execution name
        :param project_id: refer to which project
        :param properties: the properties of the workflow execution
        :param start_time: the workflow execution start time
        :param end_time: the workflow execution end time
        :param log_uri: the address of the workflow execution log
        :param workflow_json: the workflow json format
        :param signature: the signature of the code
        :param uuid: the address of the workflow execution log
        """
        self.name = name
        self.execution_state = execution_state
        self.project_id = project_id
        self.properties = properties
        self.start_time = start_time
        self.end_time = end_time
        self.log_uri = log_uri
        self.workflow_json = workflow_json
        self.signature = signature
        self.uuid = uuid

    def __str__(self):
        return '<\n' \
               'WorkflowExecutionMeta\n' \
               'uuid:{}\n' \
               'name:{},\n' \
               'project_id:{},\n' \
               'execution_state:{},\n' \
               'properties:{},\n' \
               'start_time:{},\n' \
               'end_time:{},\n' \
               'log_uri:{},\n' \
               'workflow_json:{},\n' \
               'signature:{},\n' \
               '>'.format(self.uuid,
                          self.name,
                          self.project_id,
                          self.execution_state,
                          self.properties,
                          self.start_time,
                          self.end_time,
                          self.log_uri,
                          self.workflow_json,
                          self.signature)


def create_workflow_execution(
        name: Text,
        execution_state: State,
        project_id: int,
        properties: Properties = None,
        start_time: int = None,
        end_time: int = None,
        log_uri: Text = None,
        workflow_json: Text = None,
        signature: Text = None):
    return WorkflowExecutionMeta(name=name, properties=properties, start_time=start_time, end_time=end_time,
                                 execution_state=execution_state, log_uri=log_uri, project_id=project_id,
                                 workflow_json=workflow_json,
                                 signature=signature)
